File: app/src/main/java/com/example/mvctutorial/designpattern/_16mediator/java/content/Mediator.java
package com.example.mvctutorial.designpattern._16mediator.java.content;

import java.util.ArrayList;
import java.util.List;

public abstract class Mediator {

    public List<Colleague> colleagues = new ArrayList<>();

    public boolean addColleague(Colleague colleague){
        if (colleague != null){
            return colleagues.add(colleague);
        }else{
            return false;
        }
    }

    public abstract void mediate(String data);
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_16mediator/content/Mediator.kt
package com.example.mvctutorial.designpattern._16mediator.content

import java.util.ArrayList

abstract class Mediator {
    var colleagues: MutableList<Colleague> = ArrayList()

    open fun addColleague(colleague: Colleague): Boolean {
        return colleagues.add(colleague)
    }

    abstract fun mediate(data: String)
}